src/nunit.analyzers/NonTestMethodAccessibilityLevel/NonTestMethodAccessibilityLevelAnalyzer.cs

```diff
@@ -79,5 +86,20 @@ private static bool IsPublicOrInternalMethod(IMethodSymbol method)
                     return false;
             }
         }
+
+        private static bool IsDisposeMethod(IMethodSymbol method)
+        {
+            if (method.IsOverride)
+            {
+                return IsDisposeMethod(method.OverriddenMethod);
+            }
+
+            INamedTypeSymbol? type = method.ContainingType;
+            var disposeInterfaceMethods = type.AllInterfaces
+                    .Where(x => x.SpecialType == SpecialType.System_IDisposable)
+                    .SelectMany(i => i.GetMembers().OfType<IMethodSymbol>());
+
+            return disposeInterfaceMethods.Any(m => type.FindImplementationForInterfaceMember(m).Equals(method));
```

Comment:
I think FindImplementationForInterfaceMember sometimes can return null, but perhaps this is not the case here?

Reply:
Changed code to: `Equals(type.FindImplementationForInterfaceMember(m), method)`